src/framework/preferences/js/CaptionsEnactor.js

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/* global YT */
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /*******************************************************************************
+     * captions
+     *
+     * An enactor that is capable enabling captions on embedded YouTube videos
+     *******************************************************************************/
+
+    fluid.defaults("fluid.prefs.enactor.captions", {
+        gradeNames: ["fluid.prefs.enactor", "fluid.viewComponent"],
+        preferenceMap: {
+            "fluid.prefs.captions": {
+                "model.enabled": "default"
+            }
+        },
+        events: {
+            onVideoElmLocated: null
+        },
+        selectors: {
+            videos: "iframe[src^=\"https://www.youtube.com/embed/\"]"
+        },
+        model: {
+            enabled: false
+        },
+        dynamicComponents: {
+            player: {
+                type: "fluid.prefs.enactor.captions.youTubePlayer",
+                createOnEvent: "onVideoElmLocated",
+                container: "{arguments}.0",
+                options: {
+                    model: {
+                        captions: "{captions}.model.enabled"
+                    }
+                }
+            }
+        },
+        listeners: {
+            "onCreate.checkYouTubeAPI": "fluid.prefs.enactor.captions.waitForYouTubeAPI"
+        }
+    });
+
+    // After FLUID-6148 (https://issues.fluidproject.org/browse/FLUID-6148) is complete. It should be possible to
+    // just source the dynamic components with the videos selector. The framework should at that time be able to
+    // handle the asynchrony of waiting for YT in the expander that creates the player.
+    fluid.prefs.enactor.captions.waitForYouTubeAPI = function (that) {
+        var promise = fluid.promise();
+        promise.then(function () {
+            that.locate("videos").each(function (index, elm) {
+                that.events.onVideoElmLocated.fire($(elm));
+            });
+        }, function (error) {
+            fluid.log(fluid.logLevel.WARN, error);
+        });
+
+        if (fluid.get(window, ["YT", "Player"])) {
+            promise.resolve();
+        } else {
+            // the YouTube iframe api will call onYouTubeIframeAPIReady after the api has loaded
+            window.onYouTubeIframeAPIReady = promise.resolve;
```

Comment:
Rather than blindly bung this onto a global, please make a singleton component using resolveRootSingle which mediates this interaction